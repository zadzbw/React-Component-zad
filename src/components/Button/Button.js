import React, {PropTypes, Component}from 'react';
import classNames from 'classnames';
import oneOf from '../../utils/oneOf';

import './Button.less';

const _type = {
	type: ['primary', 'ghost', 'dashed'],
	size: ['small', 'large'],
	shape: ['circle', 'circle-o']
};
const btnPrefix = 'zad-btn';

export default class Button extends Component {
	constructor(props) {
		super(props);
	}

	static propTypes = {
		type: PropTypes.string,
		size: PropTypes.string,
		shape: PropTypes.string,
		onClick: PropTypes.func
	};

	static defaultProps = {
		onClick: () => undefined
	};


	render() {
		const {type, size, shape, className, children, ...props} = this.props;

		const sizeClass = ({
				large: 'lg',
				small: 'sm',
			})[size] || '';

		const btnClass = classNames(btnPrefix, className, {
			[`${btnPrefix}-${type}`]: type && oneOf(_type.type, type),
			[`${btnPrefix}-${sizeClass}`]: size && oneOf(_type.size, size),
			[`${btnPrefix}-${shape}`]: shape && oneOf(_type.shape, shape)
		});

		return (
			<button
				{...props}
				type="button"
				className={btnClass}
			>
				<span>{children}</span>
			</button>
		);
	}
}
