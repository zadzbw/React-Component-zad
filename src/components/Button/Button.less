@import "../../style/variables";
@import "./ButtonMixin";

.zad-btn {
  .btn();
  .btn-default();

  > .fa + span, > span + .fa, > .fa + .fa {
    margin-left: 0.5em;
  }

  &-primary {
    .btn-primary();
  }

  &-ghost {
    .btn-ghost();
  }

  &-dashed {
    .btn-dashed();
  }

  &-circle {
    .btn-circle();
  }

  &-clicked:before {
    content: '';
    position: absolute;
    border-radius: inherit;
    border: 0 solid @primary-color;
    animation: buttonClick @animation-duration-slow ease-out forwards;
  }
}

@keyframes buttonClick {
  from {
    opacity: 1;
    top: -1px;
    left: -1px;
    bottom: -1px;
    right: -1px;
    border-width: 0;
  }

  to {
    opacity: 0;
    top: -6px;
    left: -6px;
    bottom: -6px;
    right: -6px;
    border-width: 6px;
  }
}