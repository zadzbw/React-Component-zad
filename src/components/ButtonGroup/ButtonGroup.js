/**
 * Created by zad on 16/12/5.
 */
import React, {PropTypes, Component}from 'react';
import classNames from 'classnames';
import isOneOf from '../../utils/isOneOf';

import './ButtonGroup.less';

const _type = {
	size: ['small', 'large']
};
const btnGroupPrefix = 'zad-btn-group';

export default class ButtonGroup extends Component {
	constructor(props) {
		super(props);
	}

	static propTypes = {
		size: PropTypes.string
	};

	render() {
		const {size, className, ...props} = this.props;

		const sizeSuffix = ({
			large: 'lg',
			small: 'sm',
		})[size];

		const btnGroupClass = classNames(btnGroupPrefix, className, {
			[`${btnGroupPrefix}-${sizeSuffix}`]: size && isOneOf(_type.size, size)
		});

		return (
			<div {...props} className={btnGroupClass}></div>
		);
	}
}
