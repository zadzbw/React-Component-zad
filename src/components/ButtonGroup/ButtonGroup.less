@import "../../style/variables";
@import "./ButtonGroupMixin";

.zad-btn-group {
  .btn-group(zad-btn);
}
