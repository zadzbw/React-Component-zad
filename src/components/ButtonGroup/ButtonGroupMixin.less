@import "../../style/variables";
@import "../Button/ButtonMixin";

.btn-group-base(@btnClassName) {
  position: relative;
  display: inline-block;
  > .@{btnClassName} {
    margin-right: 0;

    .btn-size(@btn-padding-base; @font-size-base; @btn-border-radius-base);

    &:hover, &:focus, &:active, &-primary {
      z-index: 2;
    }
  }

  &-sm > .@{btnClassName} {
    .btn-size(@btn-group-padding-sm; @font-size-sm; @btn-border-radius-sm);
  }

  &-lg > .@{btnClassName} {
    .btn-size(@btn-padding-lg; @btn-font-size-lg; @btn-border-radius-base);
  }
}

.btn-group(@btnClassName) {
  .btn-group-base(@btnClassName);

  .@{btnClassName} + .@{btnClassName} {
    margin-left: -1px;
  }

  .@{btnClassName}:not(:first-child):not(:last-child) {
    border-radius: 0;
    padding-left: 8px;
    padding-right: 8px;
  }

  .@{btnClassName}:first-child:not(:last-child) {
    border-top-right-radius: 0;
    border-bottom-right-radius: 0;
    padding-right: 8px;
  }

  .@{btnClassName}:last-child:not(:first-child) {
    border-top-left-radius: 0;
    border-bottom-left-radius: 0;
    padding-left: 8px;
  }
}
