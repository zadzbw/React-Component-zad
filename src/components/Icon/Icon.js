import React, {PropTypes, Component} from 'react';
import classNames from 'classnames';
import isOneOf from '../../utils/isOneOf';

const _type = {
  size: ['lg', '2x', '3x', '4x', '5x'],
  rotate: [90, 180, 270]
};

export default class Icon extends Component {
  static propTypes = {
    name: PropTypes.string.isRequired,
    size: PropTypes.string,
    fixedWidth: PropTypes.bool,
    spin: PropTypes.bool,
    pulse: PropTypes.bool,
    rotate: PropTypes.number
  };

  render() {
    const {name, size, fixedWidth, spin, pulse, rotate, ...props} = this.props;
    const iconClass = classNames('fa', `fa-${name}`, {
      [`fa-${size}`]: size && isOneOf(_type.size, size),
      [`fa-rotate-${rotate}`]: rotate && isOneOf(_type.rotate, rotate),
      'fa-fw': fixedWidth,
      'fa-spin': spin,
      'fa-pulse': pulse
    });
    return (
      <i {...props} className={iconClass}/>
    );
  }
}
