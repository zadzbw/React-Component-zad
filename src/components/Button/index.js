/**
 * Created by zad on 16/11/30.
 */
import Button from './Button';

export default Button;