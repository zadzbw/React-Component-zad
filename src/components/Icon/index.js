/**
 * Created by zad on 16/11/30.
 */
import Icon from './Icon';

export default Icon;