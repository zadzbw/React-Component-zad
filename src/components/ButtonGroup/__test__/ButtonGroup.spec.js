import React from 'react';
import {mount} from 'enzyme';
import {mountToJson} from 'enzyme-to-json';
import ButtonGroup from '../../ButtonGroup';
import Button from '../../Button';
import Icon from '../../Icon';

describe('ButtonGroup test', () => {
	it('normal', () => {
		const wrapper = mount(
			<ButtonGroup>
				<Button type={'primary'}><Icon name={'shield'}/>shield</Button>
				<Button>default</Button>
				<Button type={'ghost'}>ghost</Button>
				<Button type={'dashed'}>dashed</Button>
			</ButtonGroup>
		);

		expect(wrapper.find('div').hasClass('zad-btn-group')).toBeTruthy();
		expect(wrapper.find('button').get(0).classList.contains('zad-btn-primary')).toBeTruthy();
		expect(wrapper.find('button').get(0).textContent).toEqual('shield');
		expect(wrapper.find('button').get(1).classList.contains('zad-btn')).toBeTruthy();
		expect(wrapper.find('button').get(1).textContent).toEqual('default');
		expect(wrapper.find('button').get(2).classList.contains('zad-btn-ghost')).toBeTruthy();
		expect(wrapper.find('button').get(2).textContent).toEqual('ghost');
		expect(wrapper.find('button').get(3).classList.contains('zad-btn-dashed')).toBeTruthy();
		expect(wrapper.find('button').get(3).textContent).toEqual('dashed');
		expect(mountToJson(wrapper)).toMatchSnapshot();
	});

	it('size', () => {
		const wrapper = mount(
			<ButtonGroup size={'large'}>
				<Button type={'primary'}><Icon name={'shield'}/>shield</Button>
				<Button>default</Button>
				<Button type={'ghost'}>ghost</Button>
				<Button type={'dashed'}>dashed</Button>
			</ButtonGroup>
		);

		expect(wrapper.find('div').hasClass('zad-btn-group')).toBeTruthy();
		expect(wrapper.find('div').hasClass('zad-btn-group-lg')).toBeTruthy();
		expect(wrapper.find('button').get(0).classList.contains('zad-btn-primary')).toBeTruthy();
		expect(wrapper.find('button').get(0).textContent).toEqual('shield');
		expect(wrapper.find('button').get(1).classList.contains('zad-btn')).toBeTruthy();
		expect(wrapper.find('button').get(1).textContent).toEqual('default');
		expect(wrapper.find('button').get(2).classList.contains('zad-btn-ghost')).toBeTruthy();
		expect(wrapper.find('button').get(2).textContent).toEqual('ghost');
		expect(wrapper.find('button').get(3).classList.contains('zad-btn-dashed')).toBeTruthy();
		expect(wrapper.find('button').get(3).textContent).toEqual('dashed');
		expect(mountToJson(wrapper)).toMatchSnapshot();
	});
});