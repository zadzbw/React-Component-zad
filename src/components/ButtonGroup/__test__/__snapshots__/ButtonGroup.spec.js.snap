exports[`ButtonGroup test normal 1`] = `
<ButtonGroup>
  <div
    className="zad-btn-group">
    <Button
      onClick={[Function]}
      type="primary">
      <button
        className="zad-btn zad-btn-primary"
        onClick={[Function]}
        onMouseUp={[Function]}
        type="button">
        <Icon
          name="shield">
          <i
            className="fa fa-shield" />
        </Icon>
        <span>
          shield
        </span>
      </button>
    </Button>
    <Button
      onClick={[Function]}>
      <button
        className="zad-btn"
        onClick={[Function]}
        onMouseUp={[Function]}
        type="button">
        <span>
          default
        </span>
      </button>
    </Button>
    <Button
      onClick={[Function]}
      type="ghost">
      <button
        className="zad-btn zad-btn-ghost"
        onClick={[Function]}
        onMouseUp={[Function]}
        type="button">
        <span>
          ghost
        </span>
      </button>
    </Button>
    <Button
      onClick={[Function]}
      type="dashed">
      <button
        className="zad-btn zad-btn-dashed"
        onClick={[Function]}
        onMouseUp={[Function]}
        type="button">
        <span>
          dashed
        </span>
      </button>
    </Button>
  </div>
</ButtonGroup>
`;

exports[`ButtonGroup test size 1`] = `
<ButtonGroup
  size="large">
  <div
    className="zad-btn-group zad-btn-group-lg">
    <Button
      onClick={[Function]}
      type="primary">
      <button
        className="zad-btn zad-btn-primary"
        onClick={[Function]}
        onMouseUp={[Function]}
        type="button">
        <Icon
          name="shield">
          <i
            className="fa fa-shield" />
        </Icon>
        <span>
          shield
        </span>
      </button>
    </Button>
    <Button
      onClick={[Function]}>
      <button
        className="zad-btn"
        onClick={[Function]}
        onMouseUp={[Function]}
        type="button">
        <span>
          default
        </span>
      </button>
    </Button>
    <Button
      onClick={[Function]}
      type="ghost">
      <button
        className="zad-btn zad-btn-ghost"
        onClick={[Function]}
        onMouseUp={[Function]}
        type="button">
        <span>
          ghost
        </span>
      </button>
    </Button>
    <Button
      onClick={[Function]}
      type="dashed">
      <button
        className="zad-btn zad-btn-dashed"
        onClick={[Function]}
        onMouseUp={[Function]}
        type="button">
        <span>
          dashed
        </span>
      </button>
    </Button>
  </div>
</ButtonGroup>
`;
